### ISSUE TITLE
Wire up --null-assertions flag when null safety modes are enabled

### ISSUE THREAD

**Issue by jonahwilliams** [flutter/flutter#61042]
This will induce additional asserts at the boundary of migrated and non-migrated code. For web builds, it seems like this is a compiler flag. Unclear whether this is a frontend_server flag or a vm flag for mobile/desktop.

FYI @jakemac53

**Comment by jonahwilliams** [comment]
Example web impl: https://github.com/dart-lang/sdk/blob/b524f398c30ae5554dc3446e00e33f3b280e3477/pkg/dev_compiler/tool/ddb#L281

This will need to be plumbed through our generated entrypoint in the tool

**Comment by jonahwilliams** [comment]
Status: currently waiting on a dart roll

**Comment by jonahwilliams** [comment]
Waiting on https://dart-review.googlesource.com/c/sdk/+/154562 to roll through for mobile

**Comment by jonahwilliams** [comment]
Waiting on fix for https://github.com/flutter/flutter/issues/61770 to roll through to framework

**Comment by jonahwilliams** [comment]
Fix has rolled, I'm going to temporarily enable via a dart-flags argument to reduce the roll risk.

**Comment by zanderso** [comment]
@jonahwilliams IIUC we're no longer doing things this way, right?

**Comment by jonahwilliams** [comment]
I'm still working on documentation for this decision, but I don't expect to update the bug this week (as I'm out until next monday)

**Comment by jonahwilliams** [comment]
This flag is now available for Android/iOS/Web and flutter_test, and enabled for all of our framework tests by default.

Turning this on by default is extremely breaking, so that is being put on hold for now. In the spirit of the original bug, we can now use this feature to ensure that our API migrations match up with our test cases without migrating the test cases.
